### PR TITLE
docs(Operators): add inline docs to various operators G - S

### PR DESCRIPTION
**Pull request by Cmdv**
Following instructions given in this [PR](https://github.com/ReactiveX/RxJS/pull/1079) and using @justinwoo initial documentation work as an example I've added more docs to other operators.

There are some newer operators I was unable create docs for:

1. mergeAll
2. mergeMapTo
3. mergeScan
4. observerOn
5. partition
6. publishBehavior
7. publishLast
8. publishReplay
9. sampleTime

I've added placeholders for images as I was unsure if adding them now would be helpful or not.

### CODE REVIEW

**Comment by staltz** [comment]
mergeMap is flatMap ;)
And switchMap is flatMapLatest

**Comment by Cmdv** [comment]
damn it hehe, I'll get them added :+1:

**Comment by blesh** [comment]
Great work, @Cmdv! Can you please rebase this so I can merge it? Thanks!

**Comment by Cmdv** [comment]
@blesh will do, just going to add mergeMap and flatMap then I'll merge it :+1:

**Comment by Cmdv** [comment]
@blesh merged and 2 operators added :smile:

File: src/operator/ignoreElements.ts [hunk 1]
```diff
@@ -3,6 +3,14 @@ import {Operator} from '../Operator';
 import {Subscriber} from '../Subscriber';
 import {noop} from '../util/noop';
 
+/**
+ * Ignores all items emitted by the source Observable and only passes calls of onCompleted or onError.
```
Comment (staltz) [inline src/operator/ignoreElements.ts]: Note that in RxJS 5 jargon, `onCompleted` is `complete` and `onError` is `error`.

File: src/operator/isEmpty.ts [hunk 2]
```diff
@@ -2,6 +2,13 @@ import {Operator} from '../Operator';
 import {Subscriber} from '../Subscriber';
 import {Observable} from '../Observable';
 
+/**
+ * If the source Observable is empty it returns an Observable that emits true , otherwise it emits false
```
Comment (staltz) [inline src/operator/isEmpty.ts]: extra space ` ,` => `,`, and missing period `.` at the end.

File: src/operator/merge.ts [hunk 3]
```diff
@@ -2,6 +2,15 @@ import {Observable} from '../Observable';
 import {merge as mergeStatic} from './merge-static';
 import {Scheduler} from '../Scheduler';
 
+/**
+ * Flattens an Iterable of Observables into one Observable, without any transformation.
```
Comment (staltz) [inline src/operator/merge.ts]: Isn't this description for `mergeAll`? `merge` itself could be explained with "creates a result Observable which emits values from every given input Observable".

File: src/operator/retryWhen.ts [hunk 4]
```diff
@@ -9,6 +9,20 @@ import {errorObject} from '../util/errorObject';
 import {OuterSubscriber} from '../OuterSubscriber';
 import {subscribeToResult} from '../util/subscribeToResult';
 
+/**
+ * Returns an Observable that emits the same values as the source observable with the exception of an onError.
+ * An onError will cause the emission of the Throwable that cause the error to the Observable returned from
+ * notificationHandler. If that Observable calls onComplete or onError then retry will call onCompleted or onError
```
Comment (staltz) [inline src/operator/retryWhen.ts]: `onComplete` => `complete`
`onError`  => `error`

File: src/operator/share.ts [hunk 5]
```diff
@@ -2,6 +2,16 @@ import {Observable} from '../Observable';
...
+/**
+ * Returns a new Observable that multicasts (shares) the original Observable. As long as there is at least one
+ * Subscriber this Observable will be subscribed and emitting data. When all subscribers have unsubscribed it will
+ * unsubscribe from the source Observable. Because the Observable is multicasting it makes the stream `hot`.
+ * This is an alias for .publish().refCount().
+ *
+ * <img src="./img/share.png" width="100%">
+ *
+ * @returns {Observable<T>} an Observable that upon connection causes the source Observable to emit items to its Observers
+ */
```
Comment (staltz) [inline src/operator/share.ts]: This comment should be places above the `share` function, not above the `shareSubjectFactory`.

**Comment by Cmdv** [comment]
Thanks @staltz amends made

**Comment by staltz** [comment]
Good, looks good to merge to me!

**Comment by blesh** [comment]
Merged as of 04c42b3... thanks @Cmdv!!!

**Comment by Cmdv** [comment]
whoop! have to start somewhere hehe @blesh

**Pull request closed without merging**